erpnext/projects/report/delayed_tasks_summary/delayed_tasks_summary.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.utils import date_diff, nowdate
+
+def execute(filters=None):
+	columns, data = [], []
+	data = get_data(filters)
+	columns = get_columns()
+	charts = get_chart_data(data)
+	return columns, data, None, charts
+
+def get_data(filters):
+	conditions = get_conditions(filters)
+	tasks = frappe.get_all("Task", conditions, ["name", "subject", "exp_start_date", "exp_end_date", "status", "priority", "completed_on", "progress"], order_by="creation")
+	for task in tasks:
+		if task.exp_end_date:
+			if task.completed_on:
+				task.delay = date_diff(task.completed_on, task.exp_end_date)
+			else:
+				task.delay = date_diff(nowdate(), task.exp_end_date)
+	return tasks
```

Comment:
```suggestion

def get_data(filters):
	conditions = get_conditions(filters)
	tasks = frappe.get_all("Task",
		filters=conditions,
		fields=["name", "subject", "exp_start_date", "exp_end_date",
			"status", "priority", "completed_on", "progress"],
		order_by="creation"
	)

	for task in tasks:
		if task.exp_end_date:
			if task.completed_on:
				task.delay = date_diff(task.completed_on, task.exp_end_date)
			elif task.status == "Completed":
				# task is completed but completed on is not set (for older tasks)
				task.delay = 0
			else:
				# task not completed
				task.delay = date_diff(nowdate(), task.exp_end_date)
		else:
			# task has no end date, hence no delay
			task.delay = 0

	return tasks
```

1. Space out your functions better to improve readability.
2. Cover more cases
     - Task is completed but Completed On is not set (for older tasks). This will always show the task as Delayed but we can't be sure of this since the "Completed On" field was just introduced. Let's set the delay for such tasks to be 0.
     - If Task has no End Date, there is no delay
3. Check for uniform quote styles throughout the file.